lib/cosmos/dart/lib/dart_common.rb

```diff
@@ -291,7 +291,11 @@ def read_packet_from_ple(ple)
         reader.open(packet_log.filename)
         @plr_cache[packet_log.id] = reader
       end
-      return reader.read_at_offset(ple.data_offset)
+      packet = reader.read_at_offset(ple.data_offset) 
+      unless packet
+        Cosmos::Logger.error("Failed to read at offset #{ple.data_offset} (file offset: #{reader.bytes_read}) with file size #{reader.size}")
```

Comment:
This is purely informational Logging because you're still returning the nil packet. When does this happen?

Reply:
I thought it was happening, but then wasn't able to reproduce.   The message will be there if I see it again.